src/main.rs

```diff
@@ -106,12 +110,32 @@ fn main() -> Result<(), Box<dyn Error>> {
             .arg("log")
             .arg("--numstat")
             .output()?;
+
         String::from_utf8(process.stdout)?
     } else if matches.opt_present("input") {
         let path = matches.opt_str("input").unwrap();
+
         std::fs::read_to_string(path)?
+    } else if matches.opt_present("stdin") {
+        let mut input = String::new();
+        loop {
+            let mut buffer = String::new();
+            match std::io::stdin().read_line(&mut buffer) {
+                Ok(0) => {
+                    break;
+                }
+                Ok(_) => {
+                    input = input + &buffer;
+                }
+                Err(error) => {
+                    return Err(error.into());
+                }
+            }
+        }
+
+        input
     } else {
-        todo!("Read Git history from `stdin`.");
+        panic!("unknown input method");
```

Comment:
Hmm, there is the simple and less beautiful fix, which would be just a simple `unreachable!("Unexpected input method found")`, it is less beautiful, since if one would add another input method, they would have to think of this, otherwise the error would only be found during execution.

Alternatively you could create an enum:
```rust
enum InputMethod {
    File(PathBuf),
    Stdin,
    LocalRepository,
}
```
over which you could easily add a method to get the input string:
```rust
impl InputMethod {
    pub fn commit_history(self) -> String {
           match self {
                Self::File(path) => /* read path */,
                Self::Stdin => /* read stdin */,
                Self::LocalRepository => /* execute git command */,
          }
     }
}
```

I think, that would be a better solution, since it uses the type system of rust a lot better and would probably also clean up the main method.

You could also implement the `TryFrom` trait for this enum, so that you could parse the `matches` there simply. This could then be called in main like this:
```rust
let input_method: InputMethod = matches.try_into()?;
let commits = input_method.commit_history();
```

Reply:
How about moving to #19, at first?  This way, the new implementation can be moved directly to the correct module.

Reply:
Done 👍

Reply:
Okay, I am going to implement your suggestion directly in the library crate root.

Reply:
:wrench: :heavy_check_mark: